snuba/query/conditions.py

```diff
@@ -188,6 +191,33 @@ def _get_first_level_conditions(
     In the AST, the condition is a tree, so we need some additional
     logic to extract the operands of the top level AND condition.
     """
+    # Ignore the identity pattern, e.g. cond = 1 or 1 = cond since Sentry uses this
+    # format frequently (in Discover) as a workaround to current schema limitations.
+    nested_func_call = Param(
+        "nested_func",
+        FunctionCallPattern(
+            None, Or([String(BooleanFunctions.AND), String(BooleanFunctions.OR)])
```

Comment:
You _could_ make this more specific using `function`, e.g. `String(function)` but it's just a suggestion. I don't think it makes much difference besides removing the `Or`.

Reply:
Oh good point, thanks

Reply:
On further reflection, you should move all these out as globals. There's no reason to build these matchers more than once.